src/containers/forms/AssignTeamTableCell/index.tsx

```diff
@@ -0,0 +1,145 @@
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import React, { MouseEvent, useState } from 'react';
+import { connect } from 'react-redux';
+import { Button } from 'reactstrap';
+import { Store } from 'redux';
+import AssignTeamPopover, { AssignTeamPopoverProps } from '../../../components/AssignTeamPopover';
+import { ASSIGN_TEAMS, TEAMS_ASSIGNED } from '../../../constants';
+import { stopPropagationAndPreventDefault } from '../../../helpers/utils';
+import assignmentReducer, {
+  Assignment,
+  fetchAssignments,
+  getAssignmentsArrayByPlanId,
+  reducerName as assignmentReducerName,
+  resetPlanAssignments,
+} from '../../../store/ducks/opensrp/assignments';
+import organizationsReducer, {
+  getOrganizationsById,
+  Organization,
+  reducerName as organizationsReducerName,
+} from '../../../store/ducks/opensrp/organizations';
+
+reducerRegistry.register(assignmentReducerName, assignmentReducer);
+reducerRegistry.register(organizationsReducerName, organizationsReducer);
+
+/** Interface for Assign Teams cell props */
+export interface AssignTeamCellProps {
+  assignments: Assignment[]; // all the assignments for the plan-jurisdiction
+  assignmentsArray: Assignment[]; // all the assignments for the plan
+  assignButton?: React.ElementType;
+  assignPopover?: React.ElementType;
+  resetPlanAssignmentsAction: typeof resetPlanAssignments;
+  jurisdictionId: string;
+  organizationsById: { [key: string]: Organization } | null;
+  planId: string;
+}
+
+/** Component that will be rendered in IRS planning table cells
+ * showing the button and card in the Assign Teams column
+ */
+const AssignTeamTableCell = (props: AssignTeamCellProps) => {
+  const {
+    assignments,
+    assignmentsArray,
+    assignButton,
+    assignPopover,
+    resetPlanAssignmentsAction,
+    jurisdictionId,
+    organizationsById,
+    planId,
+  } = props;
+  const [isActive, setIsActive] = useState<boolean>(false);
+
+  const onPlanAssignmentButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    setIsActive(!isActive);
+  };
+
+  const onClearAssignmentsButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    const nextAssignments = assignmentsArray.filter(
+      (a: Assignment) => a.jurisdiction !== jurisdictionId
+    );
+    resetPlanAssignmentsAction({ [planId]: [...nextAssignments] });
+    setIsActive(!isActive);
+  };
+
+  const onSaveAssignmentsButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    setIsActive(!isActive);
+  };
+
+  const AssignTeamButton = (
```

Comment:
I feel like we can move this component outside of this component and provide it via props.  For two reasons:

- nothing about it means it needs to be declared here
- it would make this component (AssignTeamtableCell) more configurable and more in line with the [open/closed principle](https://dzone.com/articles/solid-principles-openclosed-principle). @cKellyDesign what do you think about this?

Reply:
Sounds good to me, I'll move it.

Reply:
moved